### ISSUE TITLE
Metric names issue tune_sklearn.tune_basesearch.TuneBaseSearchCV._format_results

### ISSUE THREAD

**Issue by zhu0619** [ray-project/tune-sklearn#116]
In `tune_sklearn/tune_basesearch.py:603`,  the way to get the column names of the given metric names aren't robust for customized scoring function and metric name, see below
`
                df[[
                    col for col in dfs[0].columns
                    if "split" in col and "test_%s" % name in col
                ]].to_numpy() for df in finished
`
If metric names are 'metric 1', 'metric 11', the result of metric 11 will also be added in 'metric 1'.
In this case, maybe use ` col.endswith("test_%s" % name)` is better.

**Comment by richardliaw** [comment]
Seems to be a duplicate of #117 ?
